mono/mini/debugger-agent.c

```diff
@@ -7755,13 +7755,18 @@ assembly_commands (int command, guint8 *p, guint8 *end, Buffer *buf)
 			mono_error_cleanup (error);
 			t = NULL;
 		} else {
-			if (info.assembly.name)
-				NOT_IMPLEMENTED;
+			if (info.assembly.name) {
+				mono_reflection_free_type_info (&info);
+				g_free (s);
+				mono_domain_set_fast (d, TRUE);
+				return ERR_INVALID_ARGUMENT;
+			}
```

Comment:
This part seems reasonable

Reply:
I discussed with Jb Evain and he will investigate on Unity debugger side. So we should just remove the assert and return a invalid argument.